entities/src/test/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategyTest.kt

```diff
@@ -367,6 +367,59 @@ class LocalEntitiesFilterStrategyTest {
         assertThat(fallthroughFilterStrategy.fellThrough, equalTo(false))
     }
 
+    @Test
+    fun `works correctly in the optimized way with property = number`() {
+        entitiesRepository.save(
+            "things",
+            Entity.New(
+                "thing1",
+                "Thing1",
+                properties = listOf("age" to "25")
+            )
+        )
+
+        entitiesRepository.save(
+            "things",
+            Entity.New(
+                "thing2",
+                "Thing2",
+                properties = listOf("age" to "30")
+            )
+        )
+
+        val scenario = Scenario.init(
+            "Secondary instance form",
+            html(
+                head(
+                    title("Secondary instance form"),
+                    model(
+                        mainInstance(
+                            t(
+                                "data id=\"create-entity-form\"",
+                                t("question"),
+                            )
+                        ),
+                        t("instance id=\"things\" src=\"jr://file-csv/things.csv\""),
+                        bind("/data/question").type("string")
+                    )
+                ),
+                body(
+                    select1Dynamic(
+                        "/data/question",
+                        "instance('things')/root/item[age=25]",
```

Comment:
I was a little concerned about only having one test here as my instinct was that we need to consider cases for both `Double` and `Int`. However, as far as I can see, JavaRosa **always** treats numbers as `Double` types for the purposes of evaluation:

1. `IntegerData` carries an `Int` around, but when a path expression is unpacked (using `#unpack`) for evaluation it gets converted to a `Double`
2. `XPathNumericLiteral` carries a `Double` around so integer literals never end up as `Int`

I was aware of 2, but 1 surprised me. Does this blanket statement around not needing to worry about `Int` seem true @lognaturel? If so, the nice thing is this single test is enough as all "numeric" data will have the same type.

Reply:
This doesn't confirm that the above is correct, but I also realized I could check Crashlytics to see if we've ever had a cash casting `Int` to `String` in `LocalEntitiesFilterStrategy` and it looks like we're clear there.